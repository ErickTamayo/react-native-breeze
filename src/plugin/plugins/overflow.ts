import { PluginFunction } from "./types";

export type PluginGroups = { overflow: "hidden" | "visible" | "scroll" };

export const pattern = /^overflow-(?<overflow>hidden|visible|scroll)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { overflow } = groups;
  return { overflow };
};
