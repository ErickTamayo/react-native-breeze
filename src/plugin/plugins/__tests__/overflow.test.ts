import { pattern, plugin, PluginGroups } from "../overflow";

describe("overflow", () => {
  it("Should match the correct pattern", () => {
    expect(pattern.exec("overflow-hidden")).toBeTruthy();
    expect(pattern.exec("overflow-visible")).toBeTruthy();
    expect(pattern.exec("overflow-scroll")).toBeTruthy();
    expect(pattern.exec("overflow-")).toBeFalsy();
    expect(pattern.exec("overflow")).toBeFalsy();
  });

  it.each<[string, PluginGroups, string]>([
    ["hidden", { overflow: "hidden" }, "hidden"],
    ["visible", { overflow: "visible" }, "visible"],
    ["scroll", { overflow: "scroll" }, "scroll"],
  ])("Should parse %s style correctly", (input, groups, expected) => {
    const theme = jest.fn();

    const result = plugin({ input, groups, theme });

    expect(result).toEqual({ overflow: expected });
  });
});
