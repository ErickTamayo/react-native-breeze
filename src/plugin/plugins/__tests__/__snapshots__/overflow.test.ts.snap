// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overflow Should parse (overflow-hidden) correctly 1`] = `
Object {
  "overflow": "hidden",
}
`;

exports[`overflow Should parse (overflow-scroll) correctly 1`] = `
Object {
  "overflow": "scroll",
}
`;

exports[`overflow Should parse (overflow-visible) correctly 1`] = `
Object {
  "overflow": "visible",
}
`;
